### PR TITLE
Added new package gpac v2.2.1

### PR DESCRIPTION
**Pull request by wmanth**
## Description

This PR brings [MP4Box](https://gpac.io) to Synology NAS.

## Checklist

- [x] Build rule `all-supported` completed successfully
- [x] New installation of package completed successfully
- [x] Package upgrade completed successfully (Manually install the package again)
- [x] Package [functionality was tested](https://github.com/SynoCommunity/spksrc/wiki/Package-Update-Policy#tests-checks)
- [ ] Any needed [documentation](https://github.com/SynoCommunity/spksrc/wiki/Create-documentation) is updated/created

### Type of change

- [ ] Bug fix
- [x] New Package
- [ ] Package update
- [ ] Includes small framework changes
- [ ] This change requires a documentation update (e.g. Wiki)

### CODE REVIEW

File: cross/gpac/Makefile [hunk 1]
```diff
@@ -0,0 +1,23 @@
+PKG_NAME = $(SPK_NAME)
+PKG_VERS = $(SPK_VERS)
```
Comment (hgy59) [inline cross/gpac/Makefile:2]: cross/gpac is first and spk/gpac depends on it.
please avoid the use of SPK_ variables in cross/gpac/Makefile

general rule:
SPK_NAME must always be lower case, but PKG_NAME matches the downloaded file and might be different to SPK_NAME

Comment (wmanth) [reply cross/gpac/Makefile]: Changes applied accordingly.  Though I would avoid declaring the same values - e.g. version and revision - at two different locations.

**Comment by mreid-tt** [comment]
@wmanth, it's advisable to avoid doing a force push for each change you make. This practice can obscure smaller alterations within a large commit, making it challenging for others to review. Moreover, if someone has cloned your branch for testing purposes, a force push will disrupt their instance, necessitating manual re-synchronization and complicating peer reviews.

**Comment by wmanth** [comment]
Basically, my PR is done. No changes pending after incorporating the previous review comment. I did just rebase my commit to latest master as no checks have been triggered before.

**Comment by mreid-tt** [comment]
@hgy59, whenever you're ready, please do let me know if you've completed your review and approval. I am ready to assist with the merge and publishing process.

**Comment by hgy59** [comment]
@mreid-tt Since it's a CLI tool, not much can go wrong.
I'll be offline for a few days, so your review is probably enough...

**Comment by mreid-tt** [comment]
Well I've not worked with CLI tools much but I've gone ahead and installed it. The one thing I noted was that I could not call the `gpac` or `MP4Box` binaries from the command prompt. I then installed one of our existing command utils and saw that I could. I suspect the `Makefile` is missing the `SPK_COMMANDS` assignment which according to [Makefile variables](https://github.com/SynoCommunity/spksrc/wiki/Makefile-variables) "_create soft links in /usr/local/{folder} for command line tools_".

I believe this is needed to make an effective CLI tool. See one of the `synocli-*` packages for examples.

**Comment by mreid-tt** [comment]
@wmanth, apologies for my earlier comment. I realise you do have a `SPK_COMMANDS` defined for `MP4Box`. I assume this is the only binary to be defined? I ask because in your description you have "_It features the MP4 reference muxer MP4Box, and many other tools_" so I was wondering where the other tools were.

Otherwise the package looks good to go.

**Review by mreid-tt — APPROVED** [review]

**Comment by mreid-tt** [comment]
@wmanth, your package has beeb published. It may take 24-48 hours to propagate through to the global cache before it shows up in the Package Center on your NAS. You can see it listed at https://synocommunity.com/package/gpac.

One thing I missed was a change-log but we can make sure to include that in a future update.

**Comment by wmanth** [comment]
@mreid-tt Thanks for your support and merging the PR! I saw the change-log setting, but due to the fact this being a new package added to the SynoCommunity repo I didn't see any reason to add a change log.

**Comment by mreid-tt** [comment]
> @mreid-tt Thanks for your support and merging the PR! I saw the change-log setting, but due to the fact this being a new package added to the SynoCommunity repo I didn't see any reason to add a change log.

You're welcome. Usually for a new package we like to put "Initial release".

**Comment by mreid-tt** [comment]
Sorry, I should have removed that but I clicked one too many times on the merge. Won't be changeable at this point unfortunately.

**Pull request merged**